webapp/src/ts/components/snackbar/snackbar.component.ts

```diff
@@ -39,39 +41,59 @@ export class SnackbarComponent implements OnInit {
 
   ngOnInit() {
     this.changeDetectorRef.detach();
-    const reduxSubscription = this.store.select(Selectors.getSnackbarContent).subscribe(content => {
-      if (!content) {
+    const reduxSubscription = this.store.select(Selectors.getSnackbarContent).subscribe((snackbarContent) => {
+      if (!snackbarContent?.message) {
+        this.message = undefined;
+        this.action = undefined;
+        this.hide();
+
+        return;
+      }
+
+      const { message, action } = snackbarContent;
+      if (this.showNextMessageTimeout) {
+        clearTimeout(this.showNextMessageTimeout);
+        this.queueShowMessage(message, action);
+
         return;
       }
 
       if (this.active) {
-        this.hide(false);
-        this.setTimeout(() => this.show(content), this.ANIMATION_DURATION);
+        this.setTimeout(() => this.resetMessage(), this.ANIMATION_DURATION);
```

Comment:
I think using `resetMessage` here is creating a race condition. If I run the dev build and keep an eye on the dev tools, throttle my network to be offline and manually sync this is what I get: 
```
action @ 13:57:15.761 SET_SNACKBAR_CONTENT (in 0.10 ms)
action {payload: {…}, type: 'SET_SNACKBAR_CONTENT'}payload: {message: 'Currently syncing…', action: undefined}type: "SET_SNACKBAR_CONTENT"[[Prototype]]: Object
<Multiple of Error replicating to remote server>
action @ 13:57:15.851 SET_SNACKBAR_CONTENT (in 0.10 ms)
action {payload: {…}, type: 'SET_SNACKBAR_CONTENT'}payload: {message: 'Sync failed. Connection timed out.', action: {…}}type: "SET_SNACKBAR_CONTENT"[[Prototype]]: Object
action @ 13:57:15.853 UPDATE_REPLICATION_STATUS (in 0.00 ms)
action @ 13:57:16.103 SET_SNACKBAR_CONTENT (in 0.10 ms)
action {payload: {…}, type: 'SET_SNACKBAR_CONTENT'}payload: {message: undefined, action: undefined}type: "SET_SNACKBAR_CONTENT"[[Prototype]]: Object
action @ 13:57:21.354 SET_SNACKBAR_CONTENT (in 0.10 ms)
action {payload: {…}, type: 'SET_SNACKBAR_CONTENT'}payload: {message: undefined, action: undefined}type: "SET_SNACKBAR_CONTENT"[[Prototype]]: Object
```

It looks like we are resetting the content immediately after we set the "sync failed" message, and then we are resetting it again after 5 seconds. 

Is there a way that we can keep what is shown in the snackbar and what we have in the store in sync? so we would only see one reset OR we would see the reset BEFORE we see the "sync failed"?